index/dlg_index.c

```diff
@@ -2380,11 +2380,7 @@ struct Mailbox *mutt_index_menu(struct MuttWindow *dlg, struct Mailbox *m_init)
       {
         if (!prereq(shared->ctx, priv->menu, CHECK_IN_MAILBOX))
           break;
-        const short c_nm_query_window_duration =
-            cs_subset_number(shared->sub, "nm_query_window_duration");
-        const bool c_nm_query_window_enable =
-            cs_subset_bool(shared->sub, "nm_query_window_enable");
-        if (!c_nm_query_window_enable && (c_nm_query_window_duration <= 0))
+        if (!nm_query_window_available())
         {
           mutt_message(_("Windowed queries disabled"));
           break;
```

Comment:
Is it worth putting the mutt_message() call inside nm_query_window_available(), rather than repeating it at each call-site?

Reply:
> repeating it at each call-site?

This duplication is removed in a later commit in this PR, which separates the new feature from a refactor so it's easier to identify root causes of breakage during a bisect. Also allows reverting the refactor while keeping the feature intact.